pkg/gds/admin.go

```diff
@@ -1006,6 +1007,133 @@ func (s *Admin) Resend(c *gin.Context) {
 	c.JSON(http.StatusOK, out)
 }
 
+// ReviewTimeline returns a list of time series records containing registration state counts by week.
+func (s *Admin) ReviewTimeline(c *gin.Context) {
+	// Go needs this constant to determine the time format
+	const timeFormat = "2006-01-02"
+	var (
+		err          error
+		in           *admin.ReviewTimelineParams
+		out          *admin.ReviewTimelineReply
+		startTime    time.Time
+		endTime      time.Time
+		earliestTime time.Time
+		weekTime     time.Time
+		numWeeks     int
+		vaspCounts   []map[string]bool
+	)
+
+	// Get request parameters
+	in = new(admin.ReviewTimelineParams)
+	if err = c.ShouldBindQuery(&in); err != nil {
+		log.Warn().Err(err).Msg("could not bind request with query params")
+		c.JSON(http.StatusBadRequest, admin.ErrorResponse(err))
+		return
+	}
+
+	// Default value for start date
+	earliestTime = time.Date(2020, time.January, 1, 0, 0, 0, 0, time.Local)
+	if in.Start != "" {
+		// Parse start date
+		if startTime, err = time.Parse(timeFormat, in.Start); err != nil {
+			log.Warn().Err(err).Msg("could not parse start date")
+			c.JSON(http.StatusBadRequest, admin.ErrorResponse(fmt.Errorf("invalid start date: %s", in.Start)))
+		}
+		// Hard limit on the earliest date to avoid making the server do unnecessary work
+		if startTime.Before(earliestTime) {
+			startTime = earliestTime
+		}
+	} else {
+		startTime = earliestTime
```

Comment:
I would recommend that the default startTime should be 1 year ago; e.g. only fetch the past year of timeline data - that will make things much easier to display on the front-end. As written, this function will continue to return more and more data as time goes on. 

This brings up another issue - the weeks are currently aligned with the difference of the end and start times. If the end time is a Thursday and the start time is a Tuesday, then the weeks will be 7 day periods Wednesday-Wednesday, which will probably introduce confusion on the front end. Instead, I propose we use the `startTime.ISOWeek()` method to get the year and week number of the start time and iterate forward until we get to the `endTime.ISOWeek()`. Here is a discussion of how to get the week start date from the ISO week: https://xferion.com/golang-reverse-isoweek-get-the-date-of-the-first-day-of-iso-week/

So what does this mean in the case where the start time is a Thursday and the end time is a Tuesday? We could do a few things:

1. Just use start/end to select the week, aligning to the nearest week that includes the specified dates
2. Change start and end params to simply specifying the number of weeks to return 
3. Filter audit records by date so they only fall into the specified time period even for partial weeks

I recommend option 1

Reply:
Option 1 sounds good, I agree that it's better if the weeks are more predictable for displaying on the front end.